src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs

```diff
@@ -48,6 +49,7 @@ public class SubscriptionStorage : IDisposable, ILowMemoryHandler
         public SubscriptionStorage(DocumentDatabase db, ServerStore serverStore)
         {
             _db = db;
+            _databaseName = db.ShardedDatabaseName ?? db.Name;
```

Comment:
can you explain why we cannot use the db.Name?

Reply:
The sharded subscription is saved using the `ShardedDatabaseName` (Northwind), the db.Name is the `Name` of `DocumentDatabase` which is Northwind$0 (or any other shard index). So when I am creating `SubscriptionStorage` for each `DocumentDatabase` of shard, I have to use the `ShardedDatabaseName` (Northwind)